cdap-docs/tools/doc-cdap-default.py

```diff
@@ -0,0 +1,434 @@
+        for item in final_items:
+            table += item.rst() + '\n'
+
+    return table
+
+def rebuild():
+    """Loads the cdap-default.xml, and rebuilds it according to these rules:
+    - First two sections: 
+        General Configuration
+        Global Configuration
+    - Remaining sections alphabetical
+    - Properties within a section alphabetical
+    - Descriptions wrapped to a 70 character line-length"""
+    
+    print 'Rebuilding cdap-default.xml'
+    items = load_defaults('', include_exclusions=True)
+    exclusions = load_exclusions()
+    in_section = True
+    defaults = {}
+    section = []
+    section_counter = 0
+    properties_counter = 0
+    for item in items:
+        if str(item.__class__) == '__main__.Section':
+            if section: # End old section
+                defaults[section_name] = section
+                section = []
+            # Start new section
+            section_counter += 1
+            section_name = item.name
+        else:
+            properties_counter += 1
+            section.append(item)
+    if section: # End old section
+        defaults[section_name] = section
+        section = []  
+    
+    print "Sections: %d Keys: %d Properties: %d" % (section_counter, len(defaults.keys()), properties_counter)
+
+    # Build XML file
+    XML_CONFIG_OPEN   = '<configuration>\n'
+    XML_SECTION_SUB   = "\n    <!-- %s -->\n\n"
+    XML_PROP_OPEN     = '    <property>\n'
+    XML_NAME_SUB      = "        <name>%s</name>\n"
+    XML_VALUE_SUB     = "        <value>%s</value>\n"
+    XML_DESCRIP_OPEN  = '        <description>\n'
+    XML_DESCRIP_SUB   = "            %s\n"
+    XML_DESCRIP_CLOSE = '        </description>\n'
+    XML_FINAL         = '        <final>true</final>\n'
+    XML_PROP_CLOSE    = '    </property>\n\n'
+    XML_CONFIG_CLOSE  = '</configuration>\n'
+
+    xml = XML_CONFIG_OPEN
+    keys = defaults.keys()
+    for key in FIRST_TWO_SECTIONS:
+        keys.remove(key)
+    keys.sort()        
+    keys = FIRST_TWO_SECTIONS + keys
+    for key in keys:
+        xml += XML_SECTION_SUB % key
+        props = defaults[key]
+        props.sort(key = lambda p: p.name)
+        for prop in props:
+            xml += XML_PROP_OPEN
+            xml += XML_NAME_SUB % prop.name
+            xml += XML_VALUE_SUB % prop.value
+            xml += XML_DESCRIP_OPEN
+            if prop.description:
+                for line in textwrap.wrap(prop.description):
+                    xml += XML_DESCRIP_SUB % line
+            else:
+                print "WARNING: No description for property %s" % prop.name
+                if prop.name in exclusions:
+                    print "but in the list of exclusions"
+            xml += XML_DESCRIP_CLOSE
+            if prop.final:
+                xml += XML_FINAL
+            xml += XML_PROP_CLOSE
+    xml += XML_CONFIG_CLOSE
+    
+    filepath = default_xml_filepath('_revised.xml')
+    f = open(filepath, 'wb')
```

Comment:
If the file exists or is open, it's just overwritten. As for error handling: I don't think it's worth the effort, because this is not a command that would be run automatically...at least, not my intention. You would run it manually and check the results. More of an editing tool. I did it to because I was getting tired trying to do it manually.